src/compiler/types.ts

```diff
@@ -9850,6 +9850,13 @@ export const commentPragmas = {
     },
 } as const;
 
+export const enum JSDocParsingMode {
+    ParseAll,
+    ParseForTypeErrors,

```

Comment:
I haven't confirmed it yet, but I also think that you need to use this mode if you want to do emit, as `see/link` may impact import preservation. But, I might be wrong there and it'll require testing. It's possible that it doesn't matter, after all, I was able to make `transpileModule` use `ParseNone` without failures.

Reply:
Thankfully I believe this isn't true, I've tested `@link` and referencing an import doesn't mean the import is kept.